### DISCUSSION TITLE
Page file structure for catch-all routes in v1

### DISCUSSION THREAD

**Discussion by earlAchromatic** [vikejs/vike#1032] [Discussion] [answered]
Trying to implement a pre v1 setup I had where I used a catch all route `/*` and a corresponding renderer that I want majority of pages to use. Then define any ones that need a different template. This is nice because I can prerender tons of pages with a single `+Page.vue`.

What's the best way to accomplish this now? I can't seem to get it to work with that same route in a `+route.ts` file either directly under `pages` (just tried it, I don't think there should be anything directly in there though) and also in a `catch-all` directory but I'm not sure what to name it.

**Comment by brillout** [comment]
Have a lookt at `/examples/react-router-v1/` and `/examples/vue-router-v1/`. Does that help?

**Reply by earlAchromatic** [reply]
I took a look and have a few follow ups. First, I prefer to use vps default router not Vue router, which those examples use (but maybe I am overlooking something). Here is a screenshot of my pre v1 setup:

![image](https://github.com/brillout/vite-plugin-ssr/assets/68669571/20b93ed7-5838-43ad-80cb-56ed94b0e835)

where the `catch-all.route.js` is shown in the window to the right. This is in the top level of `pages` and let me define the catch all template but I don't think such a thing is valid in v1. Can you advise on the preferred way to achieve this?

**Reply by brillout** [reply, marked as answer]
I believe the following should work just fine:

```bash
# most pages
pages/catch-all/+route.js # defines the catch-all '/@Id'
# other pages
pages/dental-services/+Page.js
```
